provider/keycloak_attribute_importer_identity_provider_mapper.go

```diff
@@ -0,0 +1,79 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+)
+
+func resourceKeycloakAttributeImporterIdentityProviderMapper() *schema.Resource {
+	mapperSchema := map[string]*schema.Schema{
+		"user_attribute": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "User Attribute",
+		},
+		"attribute_name": {
+			Type:          schema.TypeString,
+			Optional:      true,
+			Description:   "Attribute Name",
+			ConflictsWith: []string{"attribute_friendly_name"},
+		},
+		"attribute_friendly_name": {
+			Type:          schema.TypeString,
+			Optional:      true,
+			Description:   "Attribute Friendly Name",
+			ConflictsWith: []string{"attribute_name"},
+		},
+		"claim_name": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Claim Name",
+		},
+	}
+	genericMapperResource := resourceKeycloakIdentityProviderMapper()
+	genericMapperResource.Schema = mergeSchemas(genericMapperResource.Schema, mapperSchema)
+	genericMapperResource.Create = resourceKeycloakIdentityProviderMapperCreate(getAttributeImporterIdentityProviderMapperFromData, setAttributeImporterIdentityProviderMapperData)
+	genericMapperResource.Read = resourceKeycloakIdentityProviderMapperRead(setAttributeImporterIdentityProviderMapperData)
+	genericMapperResource.Update = resourceKeycloakIdentityProviderMapperUpdate(getAttributeImporterIdentityProviderMapperFromData, setAttributeImporterIdentityProviderMapperData)
+	return genericMapperResource
+}
+
+func getAttributeImporterIdentityProviderMapperFromData(data *schema.ResourceData, meta interface{}) (*keycloak.IdentityProviderMapper, error) {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+	rec, _ := getIdentityProviderMapperFromData(data)
+	identityProvider, err := keycloakClient.GetIdentityProvider(rec.Realm, rec.IdentityProviderAlias)
+	if err != nil {
+		return nil, handleNotFoundError(err, data)
+	}
+	rec.IdentityProviderMapper = fmt.Sprintf("%s-user-attribute-idp-mapper", identityProvider.ProviderId)
+	rec.Config = &keycloak.IdentityProviderMapperConfig{
+		UserAttribute: data.Get("user_attribute").(string),
+	}
+	if identityProvider.ProviderId == "saml" {
+		if attr, ok := data.GetOk("attribute_friendly_name"); ok {
+			rec.Config.AttributeFriendlyName = attr.(string)
+		} else if attr, ok := data.GetOk("attribute_name"); ok {
+			rec.Config.Attribute = attr.(string)
+		} else {
+			return nil, fmt.Errorf(`provider.keycloak: keycloak_attribute_importer_identity_provider_mapper: %s: neither "attribute_name" nor "attribute_friendly_name" are set`, data.Get("name").(string))
```

Comment:
There are about 15 identity providers and mappers for these providers are the same or almost same as for OIDC or SAML provider. If we're going to have a separate mapper for each IDP type there'll be about 60 different mappers which confuses more than several conditions in the same provider.

Reply:
What do you think about it?

Reply:
That's a fair point.  I didn't consider that there were so many different mappers for different identity providers.

With this in mind, what would you think about updating the error message to explain that a particular field is only required because this is a `saml` or `oidc` mapper as opposed to re-using the Terraform core error string for required fields?  This will make it more clear to the user that, while an attribute is not marked as `Required`, it is still required because of the provider id.

If you're on board with this, then I am fine with you just leaving it as-is as long as you're okay with me updating this in the future.

Reply:
Done